Doc/c-api/long.rst

```diff
@@ -571,6 +571,39 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
    .. versionadded:: 3.14
 
 
+.. c:function:: int PyLong_IsPositive(PyObject *obj)
+
+   Check if the integer object *obj* is positive (``obj > 0``).
```

Comment:
AFAICT, `obj` must *not* be NULL since otherwise it crashes (it's not the same as setting an exception). Other functions in this section mention when an input must not be NULL.

Reply:
I don't think such details are documented for other functions.

But maybe we should add instead the ``NULL`` check (like e.g. PyLong_AsNativeBytes).  IIUC, there is no rule to do so for new functions.

Reply:
Generally (unless documented otherwise), the C API has undefined behaviour if you pass in a NULL `PyObject *`. Usually it'll crash. Sometimes we `assert` that it's not NULL: that would probably be best here. But we don't need an `if`.

I advise against adding a note in the docs about NULL, since it could hint (to humans) that one *could* pass NULL to other functions.


`PyLong_AsInt32` is different: there the pointer is an “output argument” that the function fills in. An explicit note makes more sense there. Generally, for many functions like that, it's sometimes useful to run *other* effects of a function without getting the result filled in, and so some functions like this do accept NULL.

Reply:
> I advise against adding a note in the docs about NULL, since it could hint (to humans) that one could pass NULL to other functions.

I see your point here. The alternative is to specify it to other functions explicitly as well. But it's probably better to just do it in one docs PR later if needed. So I don't mind not specifying "do not pass NULL".

Reply:
Ok, no docs changes.  I'll keep this comment unresolved, in case we want handle obj==NULL case in functions as suggested.